simpleclient/src/main/java/io/prometheus/client/SimpleCollector.java

```diff
@@ -164,9 +228,10 @@ protected SimpleCollector(Builder b) {
     checkMetricName(fullname);
     if (b.help.isEmpty()) throw new IllegalStateException("Help hasn't been set.");
     help = b.help;
-    labelNames = Arrays.asList(b.labelNames);
+    labelNames = b.labelNames.length == 0 ? Collections.<String>emptyList() :
```

Comment:
This is only called on creation of a metric, it's not on a hot path so I'd not complicate things.

Reply:
LabelNames is not used in other contexts? If not, I can remove this one :)
Re the bench I'm running the single threaded to avoid the contention to make less reproducible the results (due to the contention: if I have made it faster, it would hammer more often any contended data structure)

Reply:
I need to run the benchmarks in a quite machine because I'm getting very unstable results!
I will try to run it ASAP and will write them here :)

Reply:
The one here should be created and stay alive until the process terminates.